packages/extension-api/src/extension-api.d.ts

```diff
@@ -661,9 +661,20 @@ declare module '@podman-desktop/api' {
 
   export type ProviderLinks = Link;
 
+  /**
+   * An interface to provide information about provider's images
+   * to use on Dashboard, in Resource Settings and Status Bar.
+   */
   export interface ProviderImages {
     icon?: string | { light: string; dark: string };
     logo?: string | { light: string; dark: string };
+    /**
+     * A monochrome icon ID to use in Status Bar. The icon should be defined using
+     * `icons` extension point (see details here https://podman-desktop.io/docs/extensions/developing/adding-icons).
+     *
+     * If it is undefined then the image defined by icon property is used instead.
+     */
+    iconId?: string;
```

Comment:
what's the difference between icon and iconId

I'm not sure a newcomer would understand

Reply:
The difference is one is an icon ID from icons contribution and another is relative path to an image file.

We can let to use image ID in image field, but that would further complicate the code IMO.

Icon now is defined as:

```
icon?: string | { light: string; dark: string };
```

Extensions are just using icon as a string when registering a provider instance.

For Sandbox Extension specifically I need to define a monochrome icon for the status bar and a color icon for the rest of the UI (dashboard and settings).

What do you have in mind for reusing icon property above to avoid adding new iconId property and to avoid possible compatibility issues?

Reply:
`icon` = full colour icon. `iconId` = woff (monochrome vector) icon. They would be used in different circumstances so we need both / they are independent. Agreed this is potentially confusing - but don't have a better suggestion atm.

Reply:
we do know what is provided

A custom icon use the pattern `${my-icon}` like `${kind-icon}`

so if it's using this format it'll be searched as a font icon, if not using `${}` then it's a classic icon

Reply:
if it's to provide 3 items

Then it should be in the string | { light: string; dark: string };

Like light, dark, and then a 3rd one being the fontId

Reply:
@benoitf do you mean this?
```
{
  light: string,
  dark: string, 
  id: string,
}
```

because if I just use `${iconId}` in place of icon, icon.light or icon.dark and recognize it is an id by `${}` presence. I would hit the same issue when in sandbox where I cannot have the same Red Hat logo for dark and light themes and mono icon for sandbox Provider status bar item.

Reply:
yes, the option `icon : string | {light: string; dark: string; fontId?: string}` feels more natural to me as either you choose a default icon or then you can specify a dedicated icon for light mode, dark mode or a font id to be used (also using the word font emphasis that the usage will be when we need to display text using a font)

rather than having both `icon` and `iconId` at the root level